Sources/GRPCHTTP2Core/Client/GRPCClientStreamHandler.swift

```diff
@@ -167,7 +179,27 @@ extension GRPCClientStreamHandler {
 
   func close(context: ChannelHandlerContext, mode: CloseMode, promise: EventLoopPromise<Void>?) {
     switch mode {
-    case .output, .all:
+    case .input:
+      context.close(mode: .input, promise: promise)
+
+    case .output:
```

Comment:
Can we avoid the duplication here? If we switch on the mode we can set two booleans, whether to flush and whether to close, then we can conditionally do that work after the switch.

Reply:
Maybe take another look now I've made some other changes? I get the cases are quite similar, but we're doing different things on each of them:
- succeeding a promise in `output` but forwarding it on `close` on `input` and `all`;
- "closing" on `input` means forwarding the close, on `output` it means only closing in the state machine, and on `all` means doing both

I'm not sure having several different booleans (because I don't think 2 would be enough) would make this easier to read.

Reply:
Yeah, now that we handle `input` differently it doesn't make sense to do this